executor/index_lookup_hash_join.go

```diff
@@ -294,7 +394,14 @@ func (iw *indexHashJoinInnerWorker) run(ctx context.Context, cancelFunc context.
 			joinResult.err = task.err
 			break
 		}
-		err := iw.handleTask(ctx, cancelFunc, task, joinResult, h)
+		if task.keepOuterOrder {
+			if resultCh != nil {
+				close(resultCh)
```

Comment:
`indexHashJoinInnerWorker.resultCh` is valid only when indexNestedLoopHashJoin do not need to keep order. Otherwise, it will be nil. 

When `keepOuterOrder` is true, the value of `resultCh` is assigned by `task.resultCh`. And it's closed to notice the main thread that "this task has been finished, you can fetch the next task".